dcpy/lifecycle/package/generate_metadata_assets.py

```diff
@@ -1,54 +1,64 @@
-from dcpy.models.product.dataset.metadata import Metadata, SocrataMetada
-import argparse
 from pathlib import Path
-from jinja2 import Environment, FileSystemLoader
+from jinja2 import Template
 import typer
 import subprocess
+from dcpy.models.product.dataset.metadata import Metadata
+
+DEFAULT_DATA_DICTIONARY_TEMPLATE_PATH = (
+    Path(__file__).parent / "resources" / "data_dictionary_template.jinja"
+)
 
 app = typer.Typer()
 
 
-@app.command()
-def generate_pdf_from_yml(
+@app.command("pdf_from_yml")
+def _cli_wrapper_pdf_from_yml(
     yaml_file_path: str,
     html_template_path: str,
     output_html_path: str,
     output_pdf_path: str,
-    pdf_metadata_path,
-) -> Path:
-    generate_html_from_yaml(yaml_file_path, html_template_path, output_html_path)
+) -> None:
+    generate_html_from_yaml(
+        Path(yaml_file_path),
+        Path(html_template_path),
+        Path(output_html_path),
+    )
+    generate_pdf_from_html(Path(output_html_path), Path(output_pdf_path))
+    return None
+
+
+def generate_pdf_from_html(output_html_path: Path, output_pdf_path: Path) -> Path:
     subprocess.run(
         [
             "pandoc",
             output_html_path,
             "-o",
             output_pdf_path,
-            f"--metadata-file={pdf_metadata_path}",
+            "--pdf-engine=weasyprint",
         ],
         check=True,
     )
     return Path(output_pdf_path)
 
 
-@app.command()
 def generate_html_from_yaml(
-    yaml_file_path: str, html_template_path: str, output_html_path: str
+    yaml_file_path: Path,
+    output_html_path: Path,
+    html_template_path: Path,
 ) -> Path:
-    html_output = to_html(yaml_file_path, html_template_path)
-    output_path = Path(output_html_path)
-    with open(output_path, "w") as f:
-        f.write(html_output)
-    return output_path
+    metadata = Metadata.from_path(yaml_file_path, template_vars={"var1": "value1"})
 
+    with open(html_template_path, "r") as f:
+        template_text = f.read()
+    rendered_template = Template(template_text).render({"metadata": metadata})
 
-def to_html(yaml_file_path: str, html_template_path: str) -> str:
-    env = Environment(loader=FileSystemLoader(Path(html_template_path).parent))
-    template = env.get_template(Path(html_template_path).name)
-    return template.render(metadata=get_metadata(yaml_file_path))
+    if not output_html_path.parent.exists():
+        output_html_path.parent.mkdir(parents=True)
+    with open(output_html_path, "w") as f:
+        f.write(rendered_template)
 
+    return output_html_path
 
-def get_metadata(yaml_file_path: str) -> SocrataMetada:
-    metadata = Metadata.from_path(
-        Path(yaml_file_path), template_vars={"var1": "value1"}
-    )
-    return metadata
+
+if __name__ == "__main__":
```

Comment:
should we ✂️ this and add to the dcpy.cli?

Reply:
see [comment](https://github.com/NYCPlanning/data-engineering/pull/1085#discussion_r1729396643)